src/Middleware/HttpLogging/test/TestW3CLoggingMiddleware.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    internal sealed class TestW3CLoggingMiddleware : W3CLoggingMiddleware
+    {
+
+        public TestW3CLogger Logger;
+
+        public TestW3CLoggingMiddleware(RequestDelegate next, IOptionsMonitor<W3CLoggerOptions> options, TestW3CLogger w3cLogger) : base(next, options, w3cLogger)
+        {
+            Logger = w3cLogger;
```

Comment:
I don't see why this class is necessary. Surely the test that passed in the TestW3CLogger already has access to it.

Reply:
It was from before the Middleware got its W3CLogger thru DI, I can remove it now